CIP-onchain-tx-chaining/README.md

```diff
@@ -0,0 +1,198 @@
+    , txInfoOutputs         :: [TxOut] -- ^ Transaction outputs
+    , txInfoFee             :: Value -- ^ The fee paid by this transaction.
+    , txInfoMint            :: Value -- ^ The 'Value' minted by this transaction.
+    , txInfoDCert           :: [DCert] -- ^ Digests of certificates included in this transaction
+    , txInfoWdrl            :: Map StakingCredential Integer -- ^ Withdrawals
+                                                      -- /V1->V2/: changed from assoc list to a 'PlutusTx.AssocMap'
+    , txInfoValidRange      :: POSIXTimeRange -- ^ The valid range for the transaction.
+    , txInfoSignatories     :: [PubKeyHash] -- ^ Signatures provided with the transaction, attested that they all signed the tx
+    , txInfoRedeemers       :: Map ScriptPurpose Redeemer -- ^ /Added in V2:/ a table of redeemers attached to the transaction
+    , txInfoData            :: Map DatumHash Datum -- ^ The lookup table of datums attached to the transaction
+                                                  -- /V1->V2/: changed from assoc list to a 'PlutusTx.AssocMap'
+    , txInfoId              :: TxId  -- ^ Hash of the pending transaction body (i.e. transaction excluding witnesses)
+    , txInfoChainedTxId     :: Maybe TxId  -- This CIP; hash of the chained transaction based on this transaction outputs
+    }
+```
+
+Important to note that unlike the cddl specification; the `ScriptContext` `txInfoInputs` field doesn't need to change; since the creation of the script context is only needed at phase 2 validation and the transaction hash of the prevous transaction is fully defined.
+
+## Rationale: how does this CIP achieve its goals?
+<!-- The rationale fleshes out the specification by describing what motivated the design and what led to particular design decisions. It should describe alternate designs considered and related work. The rationale should provide evidence of consensus within the community and discuss significant objections or concerns raised during the discussion.
+
+It must also explain how the proposal affects the backward compatibility of existing solutions when applicable. If the proposal responds to a CPS, the 'Rationale' section should explain how it addresses the CPS, and answer any questions that the CPS poses for potential solutions.
+-->
+
+The proposed changes allow to enforce the transaction validation process to fail in the evenience one or more of the chained transaction fails too.
+
+In particular the phase 1 validation chan check for the exsistence of an other transaction with the specified hash in the local mempool (or whait if none is present)
+
+phase 1 validation can also fail if some transaction input is defined using only an unsigned integer but no other transaction specifies the current transaction as chained.
+
+DApp developers can check on-chain for eventual chained transaction by exposing the `txInfoChainedTxId` `ScriptContext` field in the output datum going to be used in the chained transaciton.
```

Comment:
It would really help to have some usecases to see how this is useful. I think I can guess, but please just explain!